src/main/kotlin/coffeelisp/functions/TypePredicates.kt

```diff
@@ -0,0 +1,23 @@
+package coffeelisp.functions
+
+import coffeelisp.types.ConsCell
+import coffeelisp.types.Fn
+import coffeelisp.types.LispBoolean
+import coffeelisp.types.LispNumber
+import coffeelisp.types.LispString
+import coffeelisp.types.LispUnit
+import coffeelisp.types.toLispBoolean
+
+private inline fun <reified T>createTypePredicate(name: String): Fn {
+    return Fn(name) { args, env ->
+        valArgs(args, 1, name)
+        (args[0].eval(env) is T).toLispBoolean()
+    }
+}
+
+val isBoolean = createTypePredicate<LispBoolean>("boolean?")
+val isConsCell = createTypePredicate<ConsCell>("cons?")
```

Comment:
thoughts on this vs ```pair?``` and ```list?```

Reply:
I would call it `pair?` to match the Scheme standard, but that's up to your aesthetic standards :stuck_out_tongue: 

`list?` should return true if the argument is the empty list or a pair whose `cdr` is recursively a list.

If you want to implement something approaching the Scheme standard, [R5RS](https://schemers.org/Documents/Standards/R5RS/HTML/r5rs-Z-H-9.html#%_sec_6.3.2) has examples of what qualifies:

```
(pair? '(a . b))                ===>  #t
(pair? '(a b c))                ===>  #t
(pair? '())                     ===>  #f
(pair? '#(a b))                 ===>  #f
(list? '(a b c))                ===>  #t
(list? '())                     ===>  #t
(list? '(a . b))                ===>  #f
(let ((x (list 'a)))
  (set-cdr! x x)
  (list? x))                    ===>  #f
```